src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/graphs/JTBestHaplotype.java

```diff
@@ -0,0 +1,260 @@
+package org.broadinstitute.hellbender.tools.walkers.haplotypecaller.graphs;
+
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.tools.walkers.haplotypecaller.readthreading.ExperimentalReadThreadingGraph;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * A best haplotype object for being used with junction trees.
+ *
+ * Each path holds a list of all the junction trees describing its current path. This list consists of pointers to nodes
+ * in the junction trees corresponding to the paths that have already been taken by the JTBestHaplotype object.
+ *
+ * In order to invoke the junction trees simply call {@link #getApplicableNextEdgesBasedOnJunctionTrees} which will return a list
+ * of cloned path objects corresponding to each path present in the eldest tree. This method handles popping old trees with insufficient
+ * data off of the list as well as incrementing all of the trees in the list to point at the next element based on the chosen path.
+ */
+public class JTBestHaplotype<V extends BaseVertex, E extends BaseEdge> extends KBestHaplotype<V, E> {
+    private JunctionTreeView treesInQueue; // An object for storing and managing operations on the queue of junction trees active for this path
+    private int decisionEdgesTakenSinceLastJunctionTreeEvidence;
```

Comment:
I have pushed this decision into my next branch dependent on this one. It will come with the additional changes made to recover paths dropped by the junction tree traversals. For what its worth it is already implemented there.